stm/stm.ml

```diff
@@ -370,6 +321,12 @@ end = struct (* {{{ *)
   open Printf
 
   let print_dag vcs () =
+
+    (* Due to threading, this will be called from the toplevel with
+       we_are_parsing set to true, as indeed, the toplevel is waiting
+       for input . What a race! XD *)
+    Flags.we_are_parsing := false;
```

Comment:
The race is like this:
- The stm schedules the plotting in an ocaml thread, it gets queued.
- The toplevel calls `Stm.parse_sentence` with a parseable connected to `stdin`.
- `Stm.parse_sentence` sets the `we_are_parsing` flag to true.
- Reading from the console is a blocks, with the flag on, now the pending DAG drawing thread will be scheduled.
- Boom!

A good lesson on the dangers of imperative flags :D